### PR TITLE
fix(router-lite): URL generation in child component

### LINKED ISSUE #1776: [router-lite] Open in new tab does not grab correct url
# 🐛 Bug Report

On Aurelia 2 opening a link in a new tab results in the url only containing what in the load area.
IE if I am on the route /something/foo
and I have a <a load="bar">
If I just click on it I am properly navigated to /something/foo/bar
However if you open it in a new tab you are navigated to /bar

Repro link https://stackblitz.com/edit/au2-conventions?file=src%2Fmy-app.html

## 🤔 Expected Behavior

I am properly navigated to /something/foo/bar

## 😯 Current Behavior

if you open it in a new tab you are navigated to /bar

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Aurelia          |
| Language         |
| Browser          |
| Bundler          |
| Operating System |
| NPM/Node/Yarn    |

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

### 🎫 Issues

Fixes #1776

### CODE REVIEW

File: packages/router-lite/src/instructions.ts [hunk 1]
```diff
@@ -286,7 +288,7 @@ export class ViewportInstructionTree {
     }
 
     if (typeof instructionOrInstructions === 'string') {
-      const expr = RouteExpression.parse(instructionOrInstructions, routerOptions.useUrlFragmentHash);
+      const expr = RouteExpression.parse(instructionOrInstructions as string, routerOptions.useUrlFragmentHash);
```
Comment (bigopon) [inline packages/router-lite/src/instructions.ts:291]: we probably dont need the cast here

File: packages/router-lite/src/viewport-agent.ts [hunk 2]
```diff
@@ -860,7 +860,7 @@ function ensureTransitionHasNotErrored(tr: Transition): void {
   if (tr.error !== void 0 && !tr.erredWithUnknownRoute) throw tr.error;
 }
 
-const enum State {
+export const enum State {
```
Comment (bigopon) [inline packages/router-lite/src/viewport-agent.ts:863]: if this is only internal then we'll need to hide & remove this export too, similar like other const enums in our runtime.

File: packages/router-lite/src/instructions.ts [hunk 3]
```diff
@@ -330,14 +332,36 @@ export class ViewportInstructionTree {
...
 
-  public toUrl(useUrlFragmentHash: boolean = false): string {
+  public toUrl(isFinalInstruction: boolean, useUrlFragmentHash: boolean): string {
     let pathname: string;
     let hash: string;
+    let parentPath = '';
+
+    if (!isFinalInstruction) {
+      const parentPaths: string[] = [];
+      let ctx: IRouteContext | null = this.options.context as IRouteContext | null;
```
Comment (bigopon) [inline packages/router-lite/src/instructions.ts:342]: 👍

File: packages/router-lite/src/viewport-agent.ts (hunk 2 above)
Comment (Sayan751) [reply packages/router-lite/src/viewport-agent.ts:863]: Fixed it in #1779

File: packages/router-lite/src/instructions.ts (hunk 1 above)
Comment (Sayan751) [reply packages/router-lite/src/instructions.ts:291]: Thanks for noticing this. Missed cleanup. Fixed it in #1779.

**Pull request merged**